x/attribute/keeper/keeper_test.go

```diff
@@ -972,3 +1083,66 @@ func (s *KeeperTestSuite) TestPurgeAttributes() {
 		})
 	}
 }
+
+func (s *KeeperTestSuite) TestDeleteExpiredAttributes() {
+	store := s.ctx.KVStore(s.app.GetKey(types.StoreKey))
+	past := s.startBlockTime.Add(2 - time.Hour)
+	future := s.startBlockTime.Add(time.Hour)
+
+	s.Assert().NoError(s.app.NameKeeper.SetNameRecord(s.ctx, "one.expire.testing", s.user1Addr, false), "name record should save successfully")
+	s.Assert().NoError(s.app.NameKeeper.SetNameRecord(s.ctx, "two.expire.testing", s.user1Addr, false), "name record should save successfully")
+	s.Assert().NoError(s.app.NameKeeper.SetNameRecord(s.ctx, "three.expire.testing", s.user1Addr, false), "name record should save successfully")
+	s.Assert().NoError(s.app.NameKeeper.SetNameRecord(s.ctx, "four.expire.testing", s.user1Addr, false), "name record should save successfully")
+	s.Assert().NoError(s.app.NameKeeper.SetNameRecord(s.ctx, "five.expire.testing", s.user1Addr, false), "name record should save successfully")
+
+	attr1 := types.NewAttribute("one.expire.testing", s.user1, types.AttributeType_String, []byte("test1"))
+	attr1.ExpirationDate = &past
+	s.Require().NoError(s.app.AttributeKeeper.SetAttribute(s.ctx, attr1, s.user1Addr))
+	s.Require().NotNil(store.Get(types.AttributeExpireKey(attr1)))
+	s.Require().NotNil(store.Get(types.AttributeNameAddrKeyPrefix(attr1.Name, attr1.GetAddressBytes())))
+
+	attr2 := types.NewAttribute("two.expire.testing", s.user1, types.AttributeType_String, []byte("test2"))
+	attr2.ExpirationDate = &past
+	s.Require().NoError(s.app.AttributeKeeper.SetAttribute(s.ctx, attr2, s.user1Addr))
```

Comment:
This should be an error with a past dated expiration...

Reply:
I added the check to the msg_server initially and left it unchecked in the keeper.  I can add it there as well.  I know how to reconfigure the tests.

Reply:
makes sense ... if you put it in the keeper itself then it becomes contrived to test the expiration flows...

Reply:
I also added the checks to the keeper now.